internal/kafka/internal/services/quota/ams_quota_service.go

```diff
@@ -388,27 +383,55 @@ func (q amsQuotaService) DeleteQuotaForBillingModel(subscriptionID string, kafka
 // or not (subscription, false, nil)
 func (q amsQuotaService) GetSubscriptionByID(subscriptionID string) (*amsv1.Subscription, bool, *errors.ServiceError) {
-	resp, err := q.amsClient.GetSubscriptionByID(subscriptionID)
-	if resp != nil && resp.Status() == http.StatusNotFound {
-		return resp.Body(), false, nil
-	}
+	subscription, ok, err := q.amsClient.GetSubscriptionByID(subscriptionID)
 	if err != nil {
 		return nil, false, errors.NewWithCause(errors.ErrorGeneral, err, "error getting the subscriptions for id '%s'", subscriptionID)
 	}
 
-	return resp.Body(), true, nil
+	return subscription, ok, nil
 }
 
-// Checks if an organisation has a SKU entitlement and that entitlement is still active in AMS
+func (q amsQuotaService) amsBillingModelToAMSBillingModelCategory(amsBillingModel string, kafkaBillingModel config.KafkaBillingModel) (string, bool) {
+	switch true {
+	case shared.StringEqualsIgnoreCase(amsBillingModel, ocm.AMSRelatedResourceBillingModelMarketplace.String()):
+		// legacy ams billing model marketplace, this will be removed in the future and replaced with cloud specific marketplace billing models
+		// if the kafka billing model supports 'marketplace' or any cloud specific marketplace billing model, return 'marketplace'
+		if kafkaBillingModel.HasSupportForMarketplace() {
+			return ocm.AMSRelatedResourceBillingModelMarketplace.String(), true
+		}
+	case shared.StringHasPrefixIgnoreCase(amsBillingModel, fmt.Sprintf("%s-", ocm.AMSRelatedResourceBillingModelMarketplace)):
+		// ams billing model is a cloud specific marketplace, i.e. 'marketplace-aws'
+		// if the kafka billing model supports this cloud specific marketplace billing model, return 'marketplace'
+		if kafkaBillingModel.HasSupportForAMSBillingModel(amsBillingModel) {
+			return ocm.AMSRelatedResourceBillingModelMarketplace.String(), true
+		}
+	case shared.StringEqualsIgnoreCase(amsBillingModel, ocm.AMSRelatedResourceBillingModelStandard.String()):
+		if kafkaBillingModel.HasSupportForStandard() {
+			return ocm.AMSRelatedResourceBillingModelStandard.String(), true
+		}
+	}
+
+	//logger.Logger.V(10).Infof("ams related resource billing model not found for subscription %q and kafka billing model %q", subscriptionID, kafkaBillingModel.ID)
+	return "", false
+}
+
+// IsQuotaEntitlementActive checks if an organisation has a SKU entitlement and that entitlement is still active in AMS
 func (q amsQuotaService) IsQuotaEntitlementActive(kafka *dbapi.KafkaRequest) (bool, error) {
 	kafkaBillingModel, err := q.kafkaConfig.GetBillingModelByID(kafka.InstanceType, kafka.ActualKafkaBillingModel)
 	if err != nil {
 		return false, err
 	}
 
-	amsRelatedResourceBillingModel, err := q.getAMSRelatedResourceBillingModel(kafka.SubscriptionId, kafkaBillingModel)
-	if err != nil || shared.StringEmpty(amsRelatedResourceBillingModel) {
-		return false, err
+	// Using a new `err1` otherwise I would get `nil is not nil` error here since GetSubscriptionByID returns *ServiceError
```

Comment:
When I have those scenarios I usually name the error variables:
`svcErr` for ServiceError(s)
`err` for standard go errors